src/Microsoft.DotNet.Interactive.CSharpProject/RoslynWorkspaceUtilities/RoslynWorkspaceUtilities.cs

```diff
@@ -0,0 +1,332 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Text;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using static Microsoft.DotNet.Interactive.CSharpProject.RoslynWorkspaceUtilities.BuildCacheFileUtilities;
+
+namespace Microsoft.DotNet.Interactive.CSharpProject.RoslynWorkspaceUtilities;
+
+internal static class RoslynWorkspaceUtilities
+{
+    internal static BuildDataResults GetResultsFromCacheFile(string cacheFilePath)
+    {
+        CodeAnalysis.Workspace ws = null;
+        BuildProjectData buildProjectData = null;
+        CSharpParseOptions cSharpParseOptions = null;
+
+        try
+        {
+            if (!File.Exists(cacheFilePath))
+            {
+                return null;
+            }
+
+            string fileContent = File.ReadAllText(cacheFilePath);
+
+            PopulateBuildProjectData(fileContent, out buildProjectData);
+
+            ws = GetWorkspace(buildProjectData, true);
+
+            cSharpParseOptions = CreateCSharpParseOptions(buildProjectData);
+
+            return new BuildDataResults()
+            {
+                BuildProjectData = buildProjectData,
+                ProjectFilePath = cacheFilePath,
+                CacheFilePath = cacheFilePath + cacheFilenameSuffix,
+                Workspace = ws,
+                Succeeded = true,
+                CSharpParseOptions = cSharpParseOptions
+            };
+        }
+        catch (Exception)
+        {
+            return null;
+        }
+    }
+
+    internal static BuildDataResults GetResultsFromProjectFile(string csprojFilePath)
```

Comment:
This method name is misleading, since it's actually using the cache file, not the project file.

Reply:
renamed